packages/e2e-tests/specs/autosave.test.js

```diff
@@ -89,18 +95,77 @@ describe( 'autosave', () => {
 		await saveDraftWithKeyboard();
 		await page.keyboard.type( ' after save' );
 
+		// Trigger local autosave
+		await page.evaluate( () => window.wp.data.dispatch( 'core/editor' ).localAutosave() );
 		// Reload without saving on the server
-		await sleep( AUTOSAVE_INTERVAL_SECONDS + 1 );
 		await page.reload();
 
 		const notice = await page.$eval( '.components-notice__content', ( element ) => element.innerText );
-		expect( notice ).toContain( 'The backup of this post in your browser is different from the version below.' );
+		expect( notice ).toContain( AUTOSAVE_NOTICE_LOCAL );
 
 		expect( await getEditedPostContent() ).toEqual( wrapParagraph( 'before save' ) );
 		await page.click( '.components-notice__action' );
 		expect( await getEditedPostContent() ).toEqual( wrapParagraph( 'before save after save' ) );
 	} );
 
+	it( 'should clear local autosave after successful remote autosave', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'before save' );
+		await saveDraft();
+
+		await page.keyboard.type( 'after save' );
+
+		// Trigger local autosave
+		await page.evaluate( () => window.wp.data.dispatch( 'core/editor' ).localAutosave() );
+		expect( await page.evaluate( () => window.sessionStorage.length ) ).toBe( 1 );
```

Comment:
not sure how save is that check (sessionStorage), I mean other things could be using sessionStorage. did you consider using a selector or something?

Reply:
> did you consider using a selector or something?

I did, but it felt wrong to write an impure selector that wouldn't read from `state`. Maybe that's just confirmation that the use of custom store controls is wrong. :)

Reply:
Makes sense, it's not important because it's a test anyway.